test/e2e/chaos_test.go

```diff
@@ -118,38 +113,25 @@ func testKillRandomMembers(parentCtx context.Context, client crclient.Client, cl
 		}()
 
 		// The etcd cluster should eventually roll out completely
-		var previousSTSError string
-		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 30*time.Minute, true, func(ctx context.Context) (bool, error) {
-			err := client.Get(ctx, crclient.ObjectKeyFromObject(etcdSts), etcdSts)
-			if err != nil {
-				if err.Error() != previousSTSError {
-					t.Logf("failed to get statefulset %s/%s: %s", etcdSts.Namespace, etcdSts.Name, err)
-					previousSTSError = err.Error()
-				}
-				return false, nil
-			}
-			return *etcdSts.Spec.Replicas == etcdSts.Status.ReadyReplicas, nil
-		})
-		g.Expect(err).NotTo(HaveOccurred(), "etcd statefulset available replicas never converged")
-		t.Logf("etcd statefulset recovered successfully")
+		e2eutil.EventuallyObject(t, ctx, "etcd StatefulSet replicas to converge", func(ctx context.Context) (*appsv1.StatefulSet, error) {
+			sts := cpomanifests.EtcdStatefulSet(guestNamespace)
```

Comment:
n.b. it's likely good hygiene to have individual variables + spots on the heap where we keep these objects. The current approach that over-writes e.g. the `etcdSts` var during the entire test's runtime is harder to follow in my opinion, often the value of that var depends on the control flow that got you to a particular spot in the test